nl_server/store/memory.py

```diff
@@ -69,74 +54,58 @@ def __init__(self, embeddings_path: str, model: any) -> None:
     self.dataset_embeddings = torch.from_numpy(df.to_numpy()).to(torch.float)
 
   #
-  # Given a list of queries, searches the in-memory embeddings index
-  # and returns a map of candidates keyed by input queries.
+  # Given a list of query embeddings, searches the in-memory embeddings index
+  # and returns a list of candidates in the same order as original queries.
   #
-  def search_vars(self,
-                  queries: List[str],
-                  skip_topics: bool = False) -> Dict[str, Dict]:
-    query_embeddings = self.model.encode(queries, show_progress_bar=False)
-    top_k = _NUM_SV_INDEX_MATCHES
-    if skip_topics:
-      top_k = _NUM_SV_INDEX_MATCHES_WITHOUT_TOPICS
+  def vector_search(self, query_embeddings: torch.Tensor,
+                    top_k: int) -> List[vars.VarCandidates]:
     hits = semantic_search(query_embeddings,
                            self.dataset_embeddings,
                            top_k=top_k)
 
     # A map from input query -> SV DCID -> matched sentence -> score for that match
```

Comment:
update this comment according to the type change?

Reply:
Done.